src/components/reportPage/ReportPage.jsx

```diff
@@ -27,19 +27,19 @@ const ReportPage = () => {
         setProgressValue((oldValue) => oldValue + 1);
       });
     };
-    fetch();
+    if (files) fetch();
   }, []);
 
   return (
     <>
       <div className="progress-indicator-container">
         <p>
-          {progressValue} / {files.length} files processed
+          {progressValue} / {files?.length ?? 0} files processed
```

Comment:
what about `?` in `files?`

Reply:
`?.` is [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)